pkg/nvlib/info/info.go

```diff
@@ -76,3 +82,56 @@ func (i *infolib) IsTegraSystem() (bool, string) {
 
 	return false, fmt.Sprintf("%v has no 'tegra' prefix", tegraFamilyFile)
 }
+
+// UsesNVGPUModule checks whether the nvgpu module is used.
+// This kernel module is used on Tegra-based systems when using the iGPU.
+// Since some of these systems also support NVML, we use the device name
+// reported by NVML to determine whether the system is an iGPU system.
+//
+// Devices that use the nvgpu module have their device names as:
+//
+//	GPU 0: Orin (nvgpu) (UUID: 54d0709b-558d-5a59-9c65-0c5fc14a21a4)
```

Comment:
Is this naming likely to be stable? I guess we can keep adding cases to this function to adapt as things change, but it still feel "brittle" to rely solely on the name like this.

Reply:
Yes, it's definitely brittle. If you recall we were pushing for a more stable mechanism to check this when `nvml` support was added to Orin-based systems. This was the suggested workaround. We should definitely revisit it.